src/System.Management.Automation/engine/Modules/AnalysisCache.cs

```diff
@@ -665,7 +667,7 @@ private void Cleanup()
                 }
             }
 
-            if (removedSomething)
+            if (_saveCacheToDisk && removedSomething)
```

Comment:
I thought about it and perhaps over optimizing for avoiding a function call.  I'll remove.